src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -162,6 +185,10 @@ public KestrelHttpParser(IKestrelTrace log)
                                     RejectRequestLine(span);
                                 }
                             }
+                            else if (ch == 0)
+                            {
+                                RejectRequestLine(span);
```

Comment:
The problem is we need to do that in a bunch of places and this probably doesn't handle all of those cases.